Sming/SmingCore/Network/WebsocketClient.h

```diff
@@ -0,0 +1,77 @@
+/* 
+ * File:   WebsocketClient.h
+ * Author: https://github.com/hrsavla
+ *
+ * Created on August 4, 2015, 1:37 PM
+ * This Websocket Client library is ported by hrsavla into Sming from
+ * https://github.com/MORA99/Stokerbot/tree/master/Libraries/WebSocketClient 
+ * 
+ * Refactor and improve by https://github.com/avr39-ripe - Alexander V, Ribchansky
+ * 
+ */
+
+#ifndef WEBSOCKETCLIENT_H
+#define	WEBSOCKETCLIENT_H
+
+#include "../Wiring/WiringFrameworkIncludes.h"
+#include "TcpClient.h"
+#include "../Delegate.h"
+#include "URL.h"
+#include "../../Services/WebHelpers/aw-sha1.h"
+#include "../../Services/WebHelpers/base64.h"
+#include "../../Wiring/WString.h"
+#include "../Digital.h"
+#include <Network/WebsocketFrame.h>
+
+class WebsocketClient;
+
+enum wsMode
+{
+	ws_Disconnected = 0, ws_Connecting, ws_Connected
+};
+
+typedef Delegate<void(WebsocketClient& wsClient, const String message)> WebSocketClientMessageDelegate;
+typedef Delegate<void(WebsocketClient& wsClient, uint8_t* data, size_t size)> WebSocketClientBinaryDelegate;
+typedef Delegate<void(WebsocketClient& wsClient, bool  successful)> WebSocketClientDisconnectDelegate;
+typedef Delegate<void(WebsocketClient& wsClient, wsMode  Mode)> WebSocketClientConnectedDelegate;
+class WebsocketClient;
+
+class WebsocketClient: protected TcpClient
+{
+public:
+	//  TcpClient wsClient(wsOnCompleted, wsOnReceive);
+	WebsocketClient(bool autoDestruct = false); //
+	virtual ~WebsocketClient() {};
+	void setWebSocketMessageHandler(WebSocketClientMessageDelegate handler);
```

Comment:
It is not a required at the moment to have documentation for the classes. But it would be very nice if a documentation is present and I am sure other developers will be thankful. If you are looking for documentation samples take a look at HardwareTimer.h.